src/pop_protocol.erl

```diff
@@ -199,20 +187,20 @@ add_new_block(Block, CurrentTime, ProtocolData)
     %% add this block to tree_data 
     %% this can trigger errors if the block if poorly formed
     %% also fails if block is orphan or already exists
-    TD1 = blocktree:add_new_block(Block, TD0),
+    TD1 = blocktree:add_block_in_order(Block, TD0),
 
     %% current last block keeps track of the last block in the chain
     %% update it, if new block is the last one
 
 
-    CurrentLastBlock = ProtocolData#protocol_data.last_block,
-    LastBlock = resolve_fork(Block, CurrentLastBlock, TD1),
+    CurrentHeadBlock = ProtocolData#protocol_data.head_block,
+    HeadBlock = resolve_fork(Block, CurrentHeadBlock, TD1),
 
-    %% ?debugVal(CurrentLastBlock),
+    %% ?debugVal(CurrentHeadBlock),
     %% ?debugVal(Block),
-    %% ?debugVal(LastBlock),
+    %% ?debugVal(HeadBlock),
 
-    NewProtocolData = ProtocolData#protocol_data{tree_data = TD1, last_block = LastBlock},
+    NewProtocolData = ProtocolData#protocol_data{tree_data = TD1, head_block = HeadBlock},
```

Comment:
+1 to renaming last to head.
I also think it's funny because I thought you changed the behavior from last to first (head) at first 😱

Reply:
you meant the front of the longest chain 😱